src/components/views/ViewMilestone.jsx

```diff
@@ -255,7 +254,19 @@ class ViewMilestone extends Component {
                             </thead>
                             <tbody>
                               {items.map((item, i) => (
-                                <MilestoneItem name={`milestoneItem-${i}`} item={item} />
+                                <MilestoneItem
+                                  key={item.date}
+                                  name={`milestoneItem-${i}`}
+                                  item={{
+                                    date: item.date,
+                                    description: item.description,
+                                    selectedFiatType,
+                                    fiatAmount,
+                                    conversionRate,
```

Comment:
The selectedFiatType, fiatAmount and conversionRate are surely (and if not should be) defined and stored per item.